azurelinuxagent/ga/guestagent.py

```diff
@@ -0,0 +1,314 @@
+import json
+import os
+import shutil
+import time
+
+from azurelinuxagent.common.event import add_event, WALAEventOperation
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.utils import textutil
+
+from azurelinuxagent.common import logger, conf
+from azurelinuxagent.common.exception import UpdateError
+from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+from azurelinuxagent.common.version import AGENT_DIR_PATTERN, AGENT_NAME, CURRENT_VERSION
+from azurelinuxagent.ga.exthandlers import HandlerManifest
+
+AGENT_ERROR_FILE = "error.json"  # File name for agent error record
+AGENT_MANIFEST_FILE = "HandlerManifest.json"
+MAX_FAILURE = 3  # Max failure allowed for agent before declare bad agent
+
+
+class GAUpdateReportState(object):
```

Comment:
This is not best place to keep this but due to circular dependencies I kept here as this is also related to agent update and also used in this module